### PR TITLE
gnome 3.18.0 packages

### PR DESCRIPTION
**Pull request by tschoonj**
This branch represents my efforts to pull in the latest gnome 3.18.0 packages.
These are (not all of them in this PR):

- [x] glib (#44221)
- [x] pango (#44218)
- [x] atk (#44222)
- [x] libgweather (#44205)
- [x] libgit2-glib (#44229) 
- [x] zenity (Homebrew/homebrew-x11#129)
- [x] gdl (#44167) 
- [x] geocode-glib (#44184) 
- [x] gtksourceview3 (#44187)
- [x] yelp-tools and yelp-xsl (#44232)
- [x] gdk-pixbuf )#44231)
- [x] libgtop (#44228)
- [x] evince (#44227)
- [x] libsoup (#44226)
- [x] glib-networking (#44225)
- [x] gsettings-desktop-schemas (#44224)
- [x] gucharmap (#44223)
- [x] anjuta (#44169)
- [x] libxml++ (#44204)
- [x] libsigc++ (make check fails -> needs a closer look)
- [x] glibmm (depends on glib 2.46.0 in #44221), already updated to 2.46.1!
- [x] pangomm
- [x] gtkmm3
- [x] gtksourceviewmm3 (Homebrew/homebrew-x11#130)
- [x] gnome-common
- [x] at-spi2-atk
- [x] at-spi2-core
- [x] gtk+3
- [x] libpeas
- [x] pygobject3
- [x] gedit
- [x] atkmm
- [x] cairomm
- [x] gobject-introspection
- [x] gnome-icon-theme
- [x] gnome-themes-standard
- [x] vte
- [x] ghex
- [x] baobab
- [x] pixman

and probably others I am forgetting now...

Please don't accept any other gnome PRs in the meantime in order not to create any unresolved dependencies. Thanks!

### CODE REVIEW

**Comment by tschoonj** [comment]
@DomT4 could you have a look at this weird error message I am getting here?

Thanks!

Perhaps the build is timing out?

**Comment by DomT4** [comment]
If you mean Travis' error, ignore it. We're not using it as our primary CI at this point in time.

**Comment by tschoonj** [comment]
Nope I am talking about Jenkins: I get three time this error

````
brew-test-bot.xml.<init>
````

with contents:

````
Failed to read test report file /Users/brew/Jenkins/prs/version/yosemite/brew-test-bot.xml
org.dom4j.DocumentException: Invalid byte 1 of 1-byte UTF-8 sequence. Nested exception: Invalid byte 1 of 1-byte UTF-8 sequence.
	at org.dom4j.io.SAXReader.read(SAXReader.java:484)
	at org.dom4j.io.SAXReader.read(SAXReader.java:264)
	at hudson.tasks.junit.SuiteResult.parse(SuiteResult.java:123)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:282)
	at hudson.tasks.junit.TestResult.parsePossiblyEmpty(TestResult.java:228)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:163)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:146)
	at hudson.tasks.junit.TestResult.<init>(TestResult.java:122)
	at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:119)
	at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:93)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
	at hudson.remoting.UserRequest.perform(UserRequest.java:121)
	at hudson.remoting.UserRequest.perform(UserRequest.java:49)
	at hudson.remoting.Request$2.run(Request.java:326)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.sun.org.apache.xerces.internal.impl.io.MalformedByteSequenceException: Invalid byte 1 of 1-byte UTF-8 sequence.
	at com.sun.org.apache.xerces.internal.impl.io.UTF8Reader.invalidByte(UTF8Reader.java:691)
	at com.sun.org.apache.xerces.internal.impl.io.UTF8Reader.read(UTF8Reader.java:557)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityScanner.load(XMLEntityScanner.java:1743)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityScanner.scanData(XMLEntityScanner.java:1242)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanCDATASection(XMLDocumentFragmentScannerImpl.java:1654)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:3020)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:606)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:117)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:510)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:848)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1213)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:649)
	at org.dom4j.io.SAXReader.read(SAXReader.java:465)
	... 18 more
Nested exception: 
com.sun.org.apache.xerces.internal.impl.io.MalformedByteSequenceException: Invalid byte 1 of 1-byte UTF-8 sequence.
	at com.sun.org.apache.xerces.internal.impl.io.UTF8Reader.invalidByte(UTF8Reader.java:691)
	at com.sun.org.apache.xerces.internal.impl.io.UTF8Reader.read(UTF8Reader.java:557)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityScanner.load(XMLEntityScanner.java:1743)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityScanner.scanData(XMLEntityScanner.java:1242)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanCDATASection(XMLDocumentFragmentScannerImpl.java:1654)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:3020)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:606)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:117)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:510)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:848)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1213)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:649)
	at org.dom4j.io.SAXReader.read(SAXReader.java:465)
	at org.dom4j.io.SAXReader.read(SAXReader.java:264)
	at hudson.tasks.junit.SuiteResult.parse(SuiteResult.java:123)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:282)
	at hudson.tasks.junit.TestResult.parsePossiblyEmpty(TestResult.java:228)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:163)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:146)
	at hudson.tasks.junit.TestResult.<init>(TestResult.java:122)
	at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:119)
	at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:93)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
	at hudson.remoting.UserRequest.perform(UserRequest.java:121)
	at hudson.remoting.UserRequest.perform(UserRequest.java:49)
	at hudson.remoting.Request$2.run(Request.java:326)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
````

**Comment by DomT4** [comment]
Not sure. Perhaps the output of one of the errors couldn't print because of a UTF-8 error or such. @mikemcquaid may know. There are actual errors though: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/33691/version=yosemite/console

**Comment by tschoonj** [comment]
Seeing a couple of these again:

````
==> Using the sandbox
Error: /usr/local/opt/xz not present or broken
Please reinstall xz. Sorry :(
````

Also two tests failing (gtkmm and gstreamermm) since they cannot deal with glibmm's new c++11 features. Easy fix for that though.

**Comment by tschoonj** [comment]
What's the Jenkins max build time nowadays?

**Comment by DomT4** [comment]
Currently 3 hours for PRs. Normally an hour.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
The failures here are actually real failures now, rather than `xz` bugs. Full readout is [here](http://bot.brew.sh/job/Homebrew%20Pull%20Requests/33796/version=yosemite/consoleFull).

**Comment by tschoonj** [comment]
I hope these changes will fix the failures...

File: Library/Formula/gobject-introspection.rb [hunk 1]
```diff
@@ -16,7 +16,9 @@ class GobjectIntrospection < Formula
 
   depends_on "pkg-config" => :run
   depends_on "glib"
+  depends_on "cairo"
   depends_on "libffi"
+  depends_on :python
```
Comment (DomT4) [inline Library/Formula/gobject-introspection.rb]: This should be `depends_on :python if MacOS.version <= :snow_leopard`.

Comment (tschoonj) [inline Library/Formula/gobject-introspection.rb]: actually no: there was a bug in at least one of the system python's on the buildbots that causes the gobject-introspection build to fail

Comment (DomT4) [inline Library/Formula/gobject-introspection.rb]: Tim is likely to wince at the idea of enforcing our Python on such a major package, but I'll let him comment. @tdsmith

Comment (tdsmith) [inline Library/Formula/gobject-introspection.rb]: Can I take a look at that failure? Removing this dependency was a big win.

Comment (tdsmith) [inline Library/Formula/gobject-introspection.rb]: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/33685/version=mavericks/testReport/junit/brew-test-bot/mavericks/install_gobject_introspection/

Well, it's a bug somewhere.

Comment (tdsmith) [inline Library/Formula/gobject-introspection.rb]: `depends_on "python" => :build if MacOS.version == :mavericks` is probably fine for now.

Comment (tdsmith) [inline Library/Formula/gobject-introspection.rb]: We can do that after pulling to avoid having to rebottle.

**Comment by DomT4** [comment]
Just fixed this error, so the next run through should be free of this at least:

```
Error: Failed to download resource "frei0r"
Download failed: https://homebrew.bintray.com/bottles/frei0r-1.4.el_capitan.bottle.tar.gz
```

**Comment by tschoonj** [comment]
I think a `:build` only dependency for python is not enough here: I am getting this error when invoking `gobject-introspection` later on while building `at-spi2-core` (and all others :disappointed: too):

````
  GISCAN   Atspi-2.0.gir
env: /usr/local/opt/python/bin/python: No such file or directory
````

What do you think?

**Comment by tdsmith** [comment]
Ugh, of course. Let's keep it scoped to Mavericks but you're right that it can't be a build dependency.

**Comment by DomT4** [comment]
Mavericks is falling apart on this error in the `nordugrid-arc` formula:

```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:626:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/utility:253:9: error: field has incomplete type 'const Arc::Endpoint'
    _T1 first;
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:2023:27: note: in instantiation of template class 'std::__1::pair<const Arc::Endpoint, Arc::EndpointQueryingStatus>' requested here
                 decltype(__is_constructible_test(declval<_Tp>(), declval<_Args>()...))
                          ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:2072:14: note: in instantiation of template class 'std::__1::__is_constructible<false, std::__1::allocator<std::__1::pair<const Arc::Endpoint, Arc::EndpointQueryingStatus> >>' requested here
    : public __is_constructible<is_scalar<_Tp>::value || is_reference<_Tp>::value,
```

**Comment by tschoonj** [comment]
This looks like a compiler bug to me, simply because it doesn't occur with the more recent clang++ compilers or Yosemite and El Capitan. 

I suggest that I exclude Mavericks here from the formula through a `fails_with` block (although I don't know exactly how to write this. Help needed here), while at the same time I will report this to the developers of nordugrid-arc, giving them the option of working around the compiler bug for a future update.

**Comment by DomT4** [comment]
```ruby
fails_with :clang if MacOS.version < :mavericks
```

Would probably be fine.

**Comment by tschoonj** [comment]
you mean `:yosemite` right?

**Comment by DomT4** [comment]
Oops. Yes.

**Comment by tschoonj** [comment]
pushed

**Comment by DomT4** [comment]
The commit message is slightly wrong. Mavericks and lower users will still have access to the formula, it'll just be built with GCC instead of Clang there. Don't worry too much about changing it here though :)

**Comment by DomT4** [comment]
On the upside, 90% of the errors went away. On the downside, there's a slightly different one along the same lines:
```
Undefined symbols for architecture x86_64:
  "Glib::path_is_absolute(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      Arc::JDLParser::UnParse(Arc::JobDescription const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const in libaccJobDescriptionParser_la-JDLParser.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[5]: *** [libaccJobDescriptionParser.la] Error 1
make[4]: *** [install-recursive] Error 1
make[3]: *** [install-recursive] Error 1
make[2]: *** [install-recursive] Error 1
make[1]: *** [install-recursive] Error 1
make: *** [install-recursive] Error 1
==> Formula
Path: /usr/local/Library/Formula/nordugrid-arc.rb
```

**Comment by tschoonj** [comment]
I am not surprised this failed: you're mixing code compiled with g++ (nordugrid-arc) and clang++ (glibmm) here, which link to different libc++ libraries...

I assumed it wouldn't even bother starting building a bottle. This will need to be disabled completely for now.

**Comment by DomT4** [comment]
Yes, indeed. Aware of the repercussions of the choice, but wanted to try since `nordugrid-arc` already had a `fails_with :clang` block for an earlier version & the alternative is stopping everyone below Yosemite being able to access that formula suddenly and upgrade, which is an extremely blunt tool.

**Comment by tschoonj** [comment]
I think you will need to restart this, something went wrong

**Comment by DomT4** [comment]
I killed it because Yosemite crashed half way through the build & didn't come back up. It was already the next thing on the build queue.

**Comment by tschoonj** [comment]
Ah, thanks!

**Comment by tschoonj** [comment]
The Mavericks bot looks stuck to me :disappointed:

**Comment by DomT4** [comment]
Sigh. Fixed. That VM is having major trouble over the last 36 hours, but hard to do more than reboot it & start the builds again whilst we have a backlog alas.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by tschoonj** [comment]
still getting these failures on Mavericks:

````
==> brew fetch --retry --build-bottle gtkmm3 pangomm synfig
==> brew install --build-bottle gtkmm3 pangomm synfig
==> FAILED
````

````
==> brew fetch --retry --build-bottle at-spi2-core gobject-introspection
==> brew install --build-bottle at-spi2-core gobject-introspection
==> FAILED
````

````
==> brew fetch --retry --build-bottle gnome-icon-theme gtk+3
==> brew install --build-bottle gnome-icon-theme gtk+3
==> FAILED
````

But no further information as to what went wrong :cry:

**Comment by tschoonj** [comment]
@DomT4 Could you have a quick look at these weird failures?

Thanks

**Comment by tschoonj** [comment]
I noticed an hour ago that pixman was also updated a couple of days ago. Pushed it to the branch and the rebuilding showed the same failures for Mavericks though the bottles appear to have been produced.

**Comment by DomT4** [comment]
Mavericks is the only failure, and the failure only occurred because the build tripped over itself fetching bottles that hadn't yet published to be used as dependencies elsewhere.

We've pulled things that did just that along the same lines before & recently so I'm going to do the same here. If we have to selectively rebottle a few Mavericks things later, so be it, but we've yet to see big problems from ignoring these types of errors (Famous last words, probably).

**Comment by DomT4** [comment]
I won't even try and list all the commits this went in as :smile_cat:, but merged. Thanks for sticking with this PR Tom. Upgrading all the GNOME things is not an entirely fun experience, I'd imagine.

**Comment by tschoonj** [comment]
You're most welcome. It was pretty hard indeed but as always I learned a thing or two doing it :smile:

I will launch a couple of follow-up PRs in x11 and science. Could you also merge in #44438? I really need this one for the one I plan in science to go ahead.

Thanks!

**Comment by DomT4** [comment]
Sure. One sec.

**Pull request closed without merging**